pkg/metadata/inventories/inventories.go

```diff
@@ -78,15 +116,17 @@ func getCheckInstanceMetadata(checkID, configProvider string) *CheckInstanceMeta
 		checkInstanceMetadata = make(CheckInstanceMetadata)
 	}
 
-	checkInstanceMetadata["last_updated"] = lastUpdated
+	checkInstanceMetadata["last_updated"] = lastUpdated.UnixNano()
 	checkInstanceMetadata["config.hash"] = checkID
 	checkInstanceMetadata["config.provider"] = configProvider
 
 	return &checkInstanceMetadata
 }
 
 // GetPayload fills and returns the check metadata payload
-func GetPayload(ac getLoadedConfigsInterface, coll getAllInstanceIDsInterface) *Payload {
+func GetPayload(hostname string, ac autoConfigInterface, coll collectorInterface) *Payload {
+	lastPayloadSent = timeNow()
```

Comment:
Should we use the `Atomic`package here to write `lastPayloadSent` and line 170 to read it?

Reply:
I think `Atomic` only has support for basic times and this is a `time.Time` :/

Reply:
Do we need a mutex then ?

Reply:
We should add a mutex then, no ?

Reply:
Added.